src/pages/library/common.js

```diff
@@ -86,8 +174,74 @@ const FilterSection = ({ onTagFilter, labels, selectedTags, labelRenderer, listD
     }, labelsToDisplay),
     _.size(labels) > numLabelsToRender && h(Link, {
       style: { display: 'block', textAlign: 'center' },
-      onClick: () => setShowAll(!showAll)
-    }, [`See ${showAll ? 'less' : 'more'}`])
+      onClick: () => {
+        setShowAll(!showAll)
+        setFilterChanges({})
+
+        const _tagFilters = {}
+        _.forEach(label => {
+          const lowerTag = _.toLower(label)
+          _tagFilters[lowerTag] = listDataByTag[lowerTag]
+        }, labels)
+        setFilteredTags(_tagFilters)
```

Comment:
To avoid mutation we can create key/value pairs (basically tuples) and use lodash's `fromPairs` function

```suggestion
        setFilteredTags(_.flow(
          _.map(label => {
            const lowerTag = _.toLower(label)
            return [lowerTag, listDataByTag[lowerTag]]
          }),
          _.fromPairs
        )(labels))
```